frontend/src/components/api-keys/DeleteApiKeyDialog.tsx

```diff
@@ -0,0 +1,98 @@
+import { Dialog, DialogContent, DialogHeader, DialogTitle } from "@/components/ui/dialog";
+import { Button } from "@/components/ui/button";
+import { useState } from "react";
+import { useAppContext } from "@/lib/contexts/AppContext";
+import { deleteApiKey } from "@/lib/api/generated";
+
+interface DeleteApiKeyDialogProps {
+  open: boolean;
+  onClose: () => void;
+  onSuccess: () => void;
+  apiKey: {
+    id: string;
+  };
+}
+
+interface ApiErrorResponse {
+  status: number;
+  data: {
+    error: {
+      code: string;
+      message: string;
+    };
+  };
+}
+
+export function DeleteApiKeyDialog({ open, onClose, onSuccess, apiKey }: DeleteApiKeyDialogProps) {
+  const { apiClient } = useAppContext();
+  const [isDeleting, setIsDeleting] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+  
+  const handleDelete = async () => {
+    if (!apiKey.id) return;
+    
+    try {
+      setIsDeleting(true);
+      setError(null);
+      
+      const response = await deleteApiKey({
+        client: apiClient,
+        body: {
+          id: apiKey.id
+        }
+      });
+
+      if (response.status !== 200) {
+        throw new Error('Failed to delete API key');
+      }
+
+      await onSuccess();
+      onClose();
+    } catch (e) {
+      let message = 'An unexpected error occurred';
+      
+      if (e instanceof Error) {
+        message = e.message;
+      } else if (typeof e === 'object' && e !== null) {
+        const apiError = e as { response?: ApiErrorResponse };
+        message = apiError.response?.data?.error?.message || message;
```

Comment:
Use ?? instead of ||

Reply:
You also created an error extraction function, you can use it to not duplicate the code here

Reply:
@bytegen-dev did you solve those?